sockeye/encoder.py

```diff
@@ -108,6 +131,18 @@ def get_rnn_cells(self) -> List[mx.rnn.BaseRNNCell]:
         """
         raise NotImplementedError()
 
+    def get_encoded_data_length(self, data_length: mx.sym.Symbol) -> mx.sym.Symbol:
+        """
+        Returns the data lengths of the encoded sequence.
+        """
+        return data_length
+
+    def get_encoded_seq_len(self, seq_len: int) -> int:
+        """
+        Returns the size of the encoded sequence.
+        """
+        return seq_len
+
```

Comment:
That was my other thought for propagating this information.  I think it's actually more natural but requires thinking about this for all encoders when only the minority will change sequence length.  It also constitutes an API change.  If we want to support more different kinds of encoders, it could be worth that change.